microprofile/WFLY-12758_integrate_opentracing.adoc

```diff
@@ -0,0 +1,108 @@
+:idseparator:       -
+
+== Overview
+
+Currently the integration of Microprofile Opentracing is minimal as you can't configure the tracer using the WildFly Management Model.
+So we will implement a way to define several tracer configurations (Jaeger) to be used in deployments.
+One of those configuration will become a default one so that deployments that don't define which configuration to use will use this one.
+In this proposal we will only support Jaeger as Tracer implementation but it should be able to be easily extendable to support other Tracers implementations:
+* Support https://github.com/eclipse/microprofile-opentracing/blob/1.3/spec/src/main/asciidoc/microprofile-opentracing.asciidoc[Eclipse MicroProfile OpenTracing 1.3] for applications deployed in WildFly, including JAX-RS and CDI integration. Passing the TCK is required.
+* Support a concrete tracer, such as Jaeger's.
+* Allow applications to supply their own tracers via `TracerResolver`.
+* Each individual application deployed (WAR) should have its own `Tracer` instance.
+* Each WAR within an EAR should be treated as an individual WAR, having their own `Tracer` instance.
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.jboss.org/browse/WFLY-12758[WFLY-12758]
+
+=== Related Issues
+
+* https://issues.apache.org/jira/browse/EAP7-1351
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+* mailto:szhantem@redhat.com[Sultan Zhantemirov]
+
+=== Testing By
+// Put an x in the relevant field to indicate if testing will be done by Engineering or QE. 
+// Discuss with QE during the Kickoff state to decide this
+[] Engineering
+
+[X] QE
+
+=== Affected Projects or Components
+The primary integration will be happening by the updating the subsystem `microprofile-opentracing-smallrye` within https://github.com/wildfly/wildfly[WildFly].
+
+=== Other Interested Projects
+
+== Requirements
+
+* Upgrade MicroProfile Opentracing 1.3.0
+* Run any OpenTracing test in WildFly integration test suite
+** tests may require changes as this major version has breaking changes.
+* Run the Eclipse MicroProfile OpenTracing TCK without failures
+
+=== Hard Requirements
+
+Adding a new opentracing subsystem management API to create Jaeger configuration so that Jaeger tracers can be injected into deployments.
+
+=== Nice-to-Have Requirements
+
+N/A
+
+=== Non-Requirements
+
+Other tracer implementations will not be configureable from the subsystem management API.
+
+== Implementation Plan
+
+We will only cover the creation of a Jaeger tracer configuration. Other tracer implementations may come afterwards but are not currently 
+
+* Create a `jaeger-tracer` resource that define the configuration of a Jaeger tracer. This resource will expose all Jaeger Tracer configuration attributes:
+    ** `propagation`: The supported trace context propagation formats:
+        *** JAEGER: The default Jaeger trace context propagation format.
+        *** B3: The Zipkin B3 trace context propagation format.
```

Comment:
As I understood, we will be using existing jaeger-client with the Jaeger tracer. Do we need an option of setting Zipkin B3 trace context propagation format?

Reply:
Well customer might wonder why they can't integrate WildFly with Zipkin anymore

Reply:
All the requirements I have seen are to integrate with Jaeger.
Because of that, there are no plans to test with Zipkin so it shouldn't be promoted.

Reply:
@jstourac and @okotek you worked on opentracing stuff before [1], please comment on Zipkin topic

[1] https://github.com/wildfly/wildfly-proposals/pull/106 and https://github.com/wildfly/wildfly-proposals/pull/236

Reply:
This is zipkin format, this can be supported at the jaeger agent level. i'm not talking about zipking agent there.

Reply:
I understand that, one agent, but 2 possible servers (jaeger, zipkin) to consume traces

Reply:
https://www.jaegertracing.io/docs/1.15/features/#backwards-compatibility-with-zipkin